src/control/server/storage/bdev/backend_class_test.go

```diff
@@ -281,3 +275,146 @@ func TestBackend_createEmptyFile(t *testing.T) {
 		})
 	}
 }
+
+func TestBackend_createJsonFile(t *testing.T) {
+	tests := map[string]struct {
+		confIn    storage.BdevConfig
+		enableVmd bool
+		expErr    error
+		expOut    string
+	}{
+		"nvme; single ssds": {
+			confIn: storage.BdevConfig{
+				Class:      storage.BdevClassNvme,
+				DeviceList: common.MockPCIAddrs(1),
+			},
+			expOut: `
+{
```

Comment:
This doesn't seem like the right way to go about testing this... It's going to be pretty fragile and annoying to maintain. Wouldn't it be better to verify that the SPDK config builder creates the expected struct with `cmp.Diff()`? We don't really need to be testing that stdlib marshals things correctly to JSON.

Reply:
I will iterate on these tests as I add more use cases and will either add struct comparison tests or replace the current tests with those. For the moment I want to verify the exact output that will be ingested by bio/spdk.

Reply:
I think it's worth having one test that verifies the overall output written to file in the right format (i.e. making sure we're calling the json lib), but that's probably enough. If there's other logic that needs checking out, would be better to compare structs.